CMakePresets.json

```diff
@@ -0,0 +1,216 @@
+{
+  "version": 3,
+  "cmakeMinimumRequired": {
+    "major": 3,
+    "minor": 21,
+    "patch": 0
+  },
+  "configurePresets": [
+    {
+      "name": "nix-base",
+      "hidden": true,
+      "generator": "Ninja",
+      "cacheVariables": {
+        "CMAKE_CXX_COMPILER_LAUNCHER": "ccache",
+        "CMAKE_FIND_FRAMEWORK": "LAST",
+        "CMAKE_FIND_PACKAGE_PREFER_CONFIG": "ON",
+        "CMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY": "OFF",
+        "CMAKE_FIND_USE_PACKAGE_REGISTRY": "OFF",
+        "CMAKE_OSX_SYSROOT": "",
+        "VAST_ENABLE_DEVELOPER_MODE": "ON",
+        "VAST_ENABLE_LSVAST": "ON",
+        "VAST_ENABLE_MDX_REGENERATE": "ON",
+        "VAST_PLUGINS": "${sourceDir}/plugins/broker;${sourceDir}/plugins/pcap;${sourceDir}/plugins/sigma;${sourceDir}/plugins/summarize"
+      }
+    },
+    {
+      "name": "clang-base",
+      "hidden": true,
+      "cacheVariables": {
+        "CMAKE_C_COMPILER": "clang",
+        "CMAKE_CXX_COMPILER": "clang++"
+      },
+      "environment": {
+        "LDFLAGS": "-fuse-ld=lld"
+      }
+    },
+    {
+      "name": "nix-clang-debug",
+      "inherits": ["nix-base", "clang-base"],
+      "displayName": "Clang Debug Config",
+      "binaryDir": "${sourceDir}/build/clang/debug",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug",
+        "CMAKE_INSTALL_PREFIX": "${sourceDir}/install/clang/debug",
```

Comment:
The modern CMake way of specifying the install prefix is at install time via `--prefix path/to/install` rather than specifying the prefix at build time.

If we put this in our presets then we should at the very least add it to our gitignore, although I think it'd be much better to just not specify this at all.

Reply:
I don't want to change my workflow just because Kitware added another runmode to CMake. Why should I be forced to use this and type in the prefix path every time?
And besides, according to the docs you can still use `--prefix` if you want even if `CMAKE_INSTALL_PREFIX` was used during configuration.`

Adding `/install` to gitignore is a good idea though.

Reply:
Fine by me if you add it to the gitignore file.